### PR TITLE
Fluid Field Group Improvements

### PR DESCRIPTION
**Pull request by bryannielsen**
This PR addresses several issues and improvements for handling field groups within a fluid field.

- Fix bug where adding new fields to field group in fluid would not allow existing entries to accept content for that field #3741
- Fix Live Preview bugs with Fluid field group #4388 
- Add new fixed_order and order parameters to {fields} tag within fluid field group
- Default order of fields within {fields} tag is controlled by field order within the Field Group
- Add ability to use a field group's field tags outside of the {fields} tag loop

Documentation PR: https://github.com/ExpressionEngine/ExpressionEngine-User-Guide/pull/866

### CODE REVIEW

File: system/ee/ExpressionEngine/Addons/fluid_field/ft.fluid_field.php [hunk 1]
```diff
@@ -287,9 +287,13 @@ public function post_save($data)
...
                 // Existing field
-                if (strpos($key, 'field_') === 0 && (!defined('CLONING_MODE') || CLONING_MODE !== true)) {
+                if (strpos($key, 'field_') === 0) {
                     $id = str_replace('field_', '', $key);
-                    if (isset($fluid_field_data[$id])) {
+                    // If we have cloned an existing field we need to populate the fluid_field_data with its data
+                    if(defined('CLONING_MODE') && CLONING_MODE === true && !isset($fluid_field_data[$id])) {
+                        $clonedField = ee('Model')->get('fluid_field:FluidField')->filter('id', $id)->first();
+                        $group_key = 'group_' . $clonedField->group;
+                    }else if (isset($fluid_field_data[$id])) {
```
Comment (intoeetive) [inline system/ee/ExpressionEngine/Addons/fluid_field/ft.fluid_field.php]: This code is not PSR-12 compatible (needs spaces)

Comment (intoeetive) [reply system/ee/ExpressionEngine/Addons/fluid_field/ft.fluid_field.php]: @bryannielsen also there are code conflicts to resolve

File: images/uploads/index.html
Comment (TomJaeger) [inline images/uploads/index.html:1]: Should this file have been removed? @bryannielsen

Comment (bryannielsen) [reply images/uploads/index.html:1]: Nope, I think I missed it in the rebase. I'll put it back 👍

**Comment by pyrobob83** [comment]
@bryannielsen Is there anything I can do to help with this? I'm happy to test things in a vanilla build if that's helpful.

**Comment by bryannielsen** [comment]
@pyrobob83 that would be much appreciated if you're willing to do some testing!  
I think we're mostly concerned with existing installations and if this change affects the order of how fluid field groups and fields are output in templates.

**Comment by pyrobob83** [comment]
@bryannielsen FWIW, this seems to be working as I'd expect it without any adverse effects.  I made sure I had multiple field groups available to my fluid field, made sure I had several blocks setup with fully populated content in an entry and noted the presentation on the frontend.  Then I manually replaced the 3 core files in the PR - I just overwrote the entire files to makes sure I didn't screw up something:
- system/ee/ExpressionEngine/Addons/fluid_field/Service/Tag.php
- system/ee/ExpressionEngine/Addons/fluid_field/ft.fluid_field.php
- system/ee/legacy/libraries/Fluid_field_parser.php

Then re-checked the presentation and it was identical so proceeded to add a new field to my field groups and those did indeed show up and became usable in existing entries and I was able to reorder the fields inside the field groups and have the order update accordingly on the Publish screen.  As far as I can see, this is what we need, though appreciate there could be some edge-case scenarios I'm not testing that could surface issues.

**Review by matthewjohns0n — APPROVED** [review]
Pulled and tested those changes, no issues.

**Pull request merged**